app/views/layouts/_course_user_badge.html.slim

```diff
@@ -2,6 +2,7 @@ div#course-user-badge
   div.col-xs-12#course-user-progress
     - progress_bar_classes = ['progress-bar-info', 'progress-bar-striped',
                               'course-user-experience-points']
+    - course_user = course_user.calculated(:achievement_count, :experience_points)
```

Comment:
This can be part of the CourseUser model -- define it as a scope, then in the Course::Controller, load the course user using that scope.